src/Microsoft.Framework.Runtime/Servicing/ServicingBreadcrumbs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet;
+
+namespace Microsoft.Framework.Runtime.Servicing
+{
+    internal class ServicingBreadcrumbs
+    {
+        private readonly string _breadcrumbsFolder;
+
+        public ServicingBreadcrumbs()
+            : this(ResolveBreadcrumbsFolder())
+        {
+        }
+
+        internal ServicingBreadcrumbs(string breadcrumbsFolder)
+        {
+            // If the directory doesn't exist, don't create it because it
+            // needs special permissions on it
+            _breadcrumbsFolder = Directory.Exists(breadcrumbsFolder) ? breadcrumbsFolder : null; ;
```

Comment:
Also not 100% sure that we shouldn't create it. We could still try and throw if the permissions aren't correct. That would allow a tool to create the breadcrumbs folder to use this class.